### PR TITLE
Add EIP: Support ENS Name for Web3 URL

### PR DESCRIPTION
**Pull request by qizhou**
When opening a pull request to submit a new EIP, please use the suggested template: https://github.com/ethereum/EIPs/blob/master/eip-template.md

We have a GitHub bot that automatically merges some PRs. It will merge yours immediately if certain criteria are met:

 - The PR edits only existing draft PRs.
 - The build passes.
 - Your GitHub username or email address is listed in the 'author' header of all affected PRs, inside <triangular brackets>.
 - If matching on email address, the email address is the one publicly listed on your GitHub profile.

### CODE REVIEW

File: EIPS/eip-6821.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+eip: 6821
+title: Support ENS Name for Web3 URL
+description: A mapping from an ENS name to the contract address in Web3 URL
+author: Qi Zhou (@qizhou), Qiang Zhu (@qzhodl)
+discussions-to: https://ethereum-magicians.org/t/eip-6821-support-ens-name-for-web3-url/13654
+status: Draft
+type: Standards Track
+category: ERC
+created: 2023-04-02
+requires: 137, 4804
```
Comment (SamWilsn) [inline EIPS/eip-6821.md:11]: Should this standard reference https://eips.ethereum.org/EIPS/eip-634?

Comment (qizhou) [reply EIPS/eip-6821.md]: Added.  One question is that EIP-634 is in `stagnant` status, which prevents finalizing the standard.  Do you have any thoughts on that?

Comment (SamWilsn) [reply EIPS/eip-6821.md]: It's kinda unfortunate. Might be worth opening a PR to bring 634 out of stagnant.

Comment (qizhou) [reply EIPS/eip-6821.md]: https://github.com/ethereum/EIPs/pull/6841

**Review by cynic-1 — DISMISSED** [review]
LGTM

File: EIPS/eip-6821.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+## Motivation
+
+ERC-4804 defines a `web3://`-scheme RFC 2396 URI to call a smart contract either by its address or a **name** from name service.  If a **name** is specified, the standard specifies a way to resolve the contract address from the name.
+
+## Specification
+
+Given **contractName** and **chainid** from a `web3://` URI defined in ERC-4804, the protocol will find the address of the contract using the following steps:
+
+1. Find the `contentcontract` text record on ENS resolver on chain **chainid**.  Return error if the chain does not have ENS or the record is an invalid ETH address.
+2. If the `contentcontract` text record does not exist, the protocol will use the ETH address of **name**.
```
Comment (SamWilsn) [inline EIPS/eip-6821.md:27]: What does "use the ETH address of **name**" mean? The resolved address from [ERC-137's contract address interface](https://eips.ethereum.org/EIPS/eip-137#contract-address-interface)?

Comment (qizhou) [reply EIPS/eip-6821.md:27]: Yes, it is the contract address in ERC-137.

Comment (qizhou) [reply EIPS/eip-6821.md]: I have updated the text here according to your comments. Please take a look!

**Review by SamWilsn — APPROVED** [review]

**Pull request merged**